dbos_transact/dbos.py

```diff
@@ -439,6 +440,34 @@ def sleep(self, seconds: float) -> None:
         with EnterDBOSCommunicator() as ctx:
             self.sys_db.sleep(ctx.workflow_uuid, ctx.curr_comm_function_id, seconds)
 
+    def set_event(self, key: str, value: Any) -> None:
```

Comment:
This needs more assertions--it can only be called from within a workflow, while communicators soon will be callable from outside.

Reply:
Fixed.